Src/Sarif.PatternMatcher/AnalyzeCommand.cs

```diff
@@ -291,6 +293,35 @@ internal static void UpdateLevelKind(MatchExpression matchExpression)
         {
             AnalyzeContext context = base.CreateContext(options, logger, runtimeErrors, policy, filePath);
 
+            // Type or member is obsolete.
+#pragma warning disable CS0618
+            if (options.FileSizeInKilobytes != options.MaxFileSizeInKilobytes)
+            {
+                if (options.FileSizeInKilobytes != DefaultMaxFileSizeInKilobytes && options.MaxFileSizeInKilobytes != DefaultMaxFileSizeInKilobytes)
+                {
+                    string message = $"Both `--max-file-size-in-kb {options.MaxFileSizeInKilobytes}` and "
+                        + $"`--file-size-in-kb {options.MaxFileSizeInKilobytes}` were used. "
+                        + "Please remove the obsolete option `--file-size-in-kb`.";
+
+                    var innerException = new ArgumentException(message);
+
+                    ThrowExitApplicationException(context: context, ExitReason.InvalidCommandLineOption, innerException);
+                }
+                else if (options.MaxFileSizeInKilobytes != DefaultMaxFileSizeInKilobytes)
+                {
+                    context.MaxFileSizeInKilobytes = options.FileSizeInKilobytes;
```

Comment:
>FileSizeInKilobytes [](http://example.com/codeflow?start=61&length=19)

This should be MaxFileSize and the next check should be FileSize.

My reasoning: if maxFileSizeInKilobytes != default -> you should use it. Else the fileSize is different from default, so, you should use that.

Reply:
Added a test case to cover, too.